src/level/level.cpp

```diff
@@ -238,22 +247,112 @@ namespace level {
         artifacts_.remove(i);
     }
 
+    void Level::enemyShoot(entities::Shooter *s, manager::Level *levelManager){
+        Collidable *collision;
+        bool firable = true;
+        if(s->canShoot()){
+            if(s->getPosition().riga == levelManager->getPlayer()->getPosition().riga){
+                if(s->getPosition().colonna < levelManager->getPlayer()->getPosition().colonna){
+                    for(int i = (s->getPosition().colonna + 1); i <  (levelManager->getPlayer()->getPosition().colonna); i++){
+                        collision = getCollision(Position (s->getPosition().riga, i), levelManager);
+                        enums::CollisionType type = enums::CollisionType::NONE;
+                        if (collision != nullptr) type = collision->getCollisionType();
+                        if(type != enums::CollisionType::NONE){
+                            firable = false;
+                        }
+                    }
```

Comment:
Con questo stai cercando di vedere se ci sono oggetti in mezzo, quando lo sposti tutto dentro `Shooter` puoi metterlo in una funzione helper apparte.
E lo chiami qualcosa del tipo `isLineOfFireClear`